net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild

```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+EGO_PN="github.com/jedisct1/${PN}"
+
+inherit fcaps golang-build systemd user
+
+DESCRIPTION="A flexible DNS proxy, with support for encrypted DNS protocols"
+HOMEPAGE="https://github.com/jedisct1/dnscrypt-proxy"
+SRC_URI="https://${EGO_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="ISC"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+FILECAPS=( cap_net_bind_service+ep usr/bin/dnscrypt-proxy )
+PATCHES=( "${FILESDIR}"/config-full-paths-r2.patch )
+
+pkg_setup() {
+	enewgroup dnscrypt
+	enewuser dnscrypt -1 -1 /var/empty dnscrypt
+}
+
+src_compile() {
+	# Create directory structure suitable for building
+	mkdir -p "src/${EGO_PN}" || die
+	rm -r "src/${EGO_PN}" || die
+	mv "${PN}" "src/${EGO_PN}" || die
+	mv "vendor" "src/" || die
+	golang-build_src_compile
+}
+
+src_install() {
+	default
+
+	dobin dnscrypt-proxy
+
+	insinto /etc/${PN}
```

Comment:
Minor nit: please don't use `${PN}` in install paths, this reduces readability and gives false sense of security that it will continue working if you rename package in Gentoo.